src/main/java/org/ice4j/ice/Agent.java

```diff
@@ -116,17 +116,35 @@
                                             = "IceProcessingState";
 
     /**
-     *  The ScheduledExecutorService to execute planned agent termination
+     *  The ScheduledExecutorService to execute Agent's scheduled tasks
      */
-    private static final ScheduledExecutorService agentTerminationScheduler;
+    private static final ScheduledExecutorService agentTasksScheduler;
 
     static
     {
+        final ThreadFactory agentThreadFactory = new ThreadFactory()
```

Comment:
a more generic `ThreadFactory` class which takes in a name prefix and whether or not threads should be daemon threads might be useful.